### PR TITLE
feat: Create Offers

### PR DESCRIPTION
**Pull request by sunguru98**
Create offer instruction is written with CPI instructions to `public_bid` and `deposit`

### CODE REVIEW

File: listing-rewards/program/tests/create_offer.rs [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    let buyer_wsol_token_account = get_associated_token_address(buyer_pubkey, &mint);
+
+    transfer_lamports(
+        &mut context,
+        &buyer,
+        &buyer_wsol_token_account,
+        listing_rewards_test::ONE_SOL + 500000000,
+    )
+    .await
+    .unwrap();
```
Comment (kespinola) [inline listing-rewards/program/tests/create_offer.rs:231]: auction house doesn't actually make use of wSOL. Its a special condition to trigger the use of sol so you can airdrop directly to the buyer's wallet. To further simplify you can make the context.payer setup by program test as the buyer.

File: listing-rewards/program/tests/create_offer.rs [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    let tx = Transaction::new_signed_with_payer(
+        &[create_offer_ix],
+        Some(buyer_pubkey),
+        &[&buyer],
+        context.last_blockhash,
+    );
+
+    let tx_response = context.banks_client.process_transaction(tx).await;
+
+    println!("{:?}", tx_response);
```
Comment (kespinola) [inline listing-rewards/program/tests/create_offer.rs:276]: drop println

File: listing-rewards/program/tests/listing_rewards_test/mod.rs [hunk 3]
```diff
@@ -4,6 +4,7 @@ use solana_program_test::*;
 
 pub const TEN_SOL: u64 = 10_000_000_000;
 pub const ONE_SOL: u64 = 1_000_000_000;
+pub const TEST_COLLECTION: &str = "Cehzo7ugAvuYcTst9HF24ackLxnrpDkzHFajj17FuyUR";
```
Comment (kespinola) [inline listing-rewards/program/tests/listing_rewards_test/mod.rs:7]: Oh test collection is already in the testing module. Drop it from program constants.

**Review by kespinola — CHANGES_REQUESTED** [review]
minor clean recommendations but looks really good. ty.

File: listing-rewards/program/src/offers/create_offer/mod.rs [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+        escrow_payment_bump,
+        buyer_price,
+        token_size,
+    )?;
+
+    mpl_auction_house::cpi::auctioneer_deposit(
+        deposit_accounts_ctx,
+        escrow_payment_bump,
+        buyer_price,
+    )?;
```
[the remark this answers is not in the document]
Comment (sunguru98) [reply listing-rewards/program/src/offers/create_offer/mod.rs:289]: Actually, I did do the suggested action, however it ended with up an error stating as ProgramFailedToExecute error. Swapping them both fixed that weird error though.

File: listing-rewards/program/src/constants.rs [hunk 5]
```diff
@@ -2,4 +2,8 @@ pub const REWARD_CENTER: &str = "reward_center";
 
 pub const REWARDABLE_COLLECTION: &str = "rewardable_collection";
 
+pub const TEST_COLLECTION: &str = "Cehzo7ugAvuYcTst9HF24ackLxnrpDkzHFajj17FuyUR";
```
[the remark this answers is not in the document]
Comment (sunguru98) [reply listing-rewards/program/src/constants.rs:5]: This obv is a blunder haha. Will delete it

**Review by kespinola — APPROVED** [review]

**Pull request merged**